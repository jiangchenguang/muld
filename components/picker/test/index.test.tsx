/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Picker from '../index';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

const Wrapper = (props: any) => {
    const simpleColumn = ['1990', '1991', '1992', '1993', '1994', '1995'];

    return <Picker {...props} columns={simpleColumn} />;
};

const MP = '.mul-picker';
const MPC = '.mul-picker-column';

it('toolbar', () => {
    afterEach(cleanup);
    const titleText = '标题123';
    const cancelText = '取消123';
    const confirmText = '确认123';
    const showToolbar = true;
    const { container } = render(
        <Wrapper
            title={titleText}
            showToolbar={showToolbar}
            confirmButtonText={confirmText}
            cancelButtonText={cancelText}
        />,
    );
    const title = container.querySelector(`${MP}__title`);
    expect(title?.textContent).toBe('标题123');
    const cancelBtn = container.querySelector(`${MP}__cancel`);
    const confirmBtn = container.querySelector(`${MP}__confirm`);
    expect(cancelBtn?.textContent).toBe('取消123');
    expect(confirmBtn?.textContent).toBe('确认123');
});

it('confirm event', () => {
    afterEach(cleanup);
    const onConfirm = jest.fn();
    const { container } = render(<Wrapper onConfirm={onConfirm} />);
    const confirmBtn = container.querySelector(`button${MP}__confirm`);
    expect(confirmBtn?.textContent).toBe('确认');
    fireEvent.click(confirmBtn!);
    expect(onConfirm).toHaveBeenCalled();
});

it('cancel event', () => {
    afterEach(cleanup);
    const onCancel = jest.fn();
    const { container } = render(<Wrapper onCancel={onCancel} />);
    const cancelBtn = container.querySelector(`button${MP}__cancel`);
    expect(cancelBtn?.textContent).toBe('取消');
    fireEvent.click(cancelBtn!);
    expect(onCancel).toHaveBeenCalled();
});

it('default index', () => {
    afterEach(cleanup);
    const defaultIndex = 2;
    const { container } = render(<Wrapper defaultIndex={defaultIndex} />);
    const columnItems = container.querySelector(`${MPC}__item--selected`);
    expect(columnItems?.textContent).toBe('1992');
});

it('render default index', () => {
    const defaultIndex = 2;
    const { asFragment } = render(<Wrapper defaultIndex={defaultIndex} />);
    expect(asFragment()).toMatchSnapshot();
});

it('render mul options', () => {
    const column3 = {
        浙江: ['杭州', '宁波', '温州', '嘉兴', '湖州'],
        福建: ['福州', '厦门', '莆田', '三明', '泉州'],
    };
    const defaultIndex = 2;
    const { asFragment } = render(<Wrapper defaultIndex={defaultIndex} columns={column3} />);
    expect(asFragment()).toMatchSnapshot();
});

it('render default slot', () => {
    const defaultSlot = <div>header</div>;
    const { asFragment } = render(<Wrapper>{defaultSlot}</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
});

it('render title slot', () => {
    const titleSlot = <div slot="title">title</div>;
    const { asFragment } = render(<Wrapper>{titleSlot}</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
});

it('render columns-top slot', () => {
    const columnTop = <div slot="columns-top">columns-top</div>;
    const { asFragment } = render(<Wrapper>{columnTop}</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
});

it('render columns-bottom slot', () => {
    const columnTop = <div slot="columns-top">columns-top</div>;
    const { asFragment } = render(<Wrapper>{columnTop}</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
});

// todo disabled
// test('disabled index', () => {
//     afterEach(cleanup);
//     const columns = [
//         { text: '杭州' },
//         { text: '宁波' },
//         { text: '温州', disabled: true },
//         { text: '嘉兴', disabled: true },
//     ];
//     const { container } = render(<Wrapper columns={columns}/>);
//     const item = container.querySelectorAll(`${MPC}__item`);
//     expect(item.length).toBe(2);
//     expect(item[0]?.textContent).toBe('温州');
//     expect(item[1]?.textContent).toBe('嘉兴');
// });
