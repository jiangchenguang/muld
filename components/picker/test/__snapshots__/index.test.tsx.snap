// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render columns-bottom slot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div
      slot="columns-top"
    >
      columns-top
    </div>
    <div
      slot="columns-top"
    >
      columns-top
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 110px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render columns-top slot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div
      slot="columns-top"
    >
      columns-top
    </div>
    <div
      slot="columns-top"
    >
      columns-top
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 110px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render default index 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div
      class="mul-picker__toolbar"
    >
      <button
        class="mul-picker__cancel"
        type="button"
      >
        取消
      </button>
      <div
        class="mul-picker__title mul-ellipsis"
      >
        标题
      </div>
      <button
        class="mul-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 22px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render default slot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div>
      header
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 110px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render mul options 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div
      class="mul-picker__toolbar"
    >
      <button
        class="mul-picker__cancel"
        type="button"
      >
        取消
      </button>
      <div
        class="mul-picker__title mul-ellipsis"
      >
        标题
      </div>
      <button
        class="mul-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 22px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render title slot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cuCmLo mul-picker"
  >
    <div
      slot="title"
    >
      title
    </div>
    <div
      class="mul-picker__columns"
      style="height: 264px;"
    >
      <div
        class="mul-picker-column"
      >
        <ul
          style="transform: translate3d(0, 110px, 0); transition-duration: 200ms; transition-property: all;"
        >
          <div>
            <li
              class="undefined mul-picker-column__item mul-picker-column__item--selected"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1990
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1991
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1992
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1993
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1994
              </div>
            </li>
          </div>
          <div>
            <li
              class="undefined mul-picker-column__item"
              role="button"
              style="height: 44px;"
              tabindex="0"
            >
              <div
                class="mul-ellipsis"
              >
                1995
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="mul-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="mul-hairline-unset--top-bottom mul-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
